### PR TITLE
update react-native-popover-view

### PR DESCRIPTION
**Pull request by drewvolz**
- This targets #6719 
- Popover v5.1.7 fixes crashes that I saw on v4.0. 
- V5.x [upgrade guide from the popover ](https://github.com/SteffeyDev/react-native-popover-view#4x-to-50)

## Issue tracking
  - [x] 6e7fae2b update `react-native-popover-view` to 5.0.0 in filter module
  - [x] 24efc8b The width does not contain the content container size
  - [x] 693f309 Add safearea insets to mind the notch and bottom bar
  - [x] b497b97 No view presented when the popover is partially off-screen on the left (solved with 5.0.0~>5.1.7)
  
## Visual fixes

Scenario | Before | After
--|--|--
Safe area  | Simulator Screen Shot - iPhone 14 Pro - 2023-01-17 at 20 32 28 | Simulator Screen Shot - iPhone 14 Pro - 2023-01-17 at 20 33 13
Content overflowing | Simulator Screen Shot - iPhone 14 Pro - 2023-01-17 at 20 33 30 | Simulator Screen Shot - iPhone 14 Pro - 2023-01-17 at 20 33 48

## Last big issue 

Fixed! Solved with upgrading grom 5.0.0 ~> 5.1.7

<details>
<summary>See more info...</summary>

Out last issue is: "no view presented when the popover is partially off-screen on the left". Tapping a filter toolbar button that's partially hidden on the left of the screen can freeze up the view (nothing presented)

Expectation | Actuality
--|--
  | 

### Debugging
This last one is tricky.
- The package does provide a `debug={true}` param which gives us some data to look at.
- There is [a troubleshooting section on the readme](https://github.com/SteffeyDev/react-native-popover-view#troubleshooting)
 
**Repro**: Current logging when opening the filter toolbar button from the far left, partially obscured.
 ```
LOG  [2023-01-18T04:13:44.854Z] calculateRectFromRef - waiting for ref
LOG  [2023-01-18T04:13:44.854Z] calculateRectFromRef - waiting for ref to move
 ```
 
I've tried [looking at open issues](https://github.com/SteffeyDev/react-native-popover-view/search?q=calculateRectFromRef+-+waiting+for+ref+to+move&type=issues) related to these logs. The idea that our ref isn't available and we're in an odd position on the far left makes me think bounds of the ref's rect could be an issue?
 
**Repro**: A normal log of opening the popover

<details>
<summary>Expand to see the larger log contents…</summary>

```
LOG  [2023-01-18T04:28:40.637Z] calculateRectFromRef - waiting for ref
LOG  [2023-01-18T04:28:40.637Z] calculateRectFromRef - waiting for ref to move
LOG  [2023-01-18T04:28:40.654Z] calculateRectFromRef - calculated Rect: {"x":10,"y":159.99999872843426,"width":152.3333282470703,"height":31.333332061767578}
LOG  [2023-01-18T04:28:40.656Z] [AdaptivePopover] getDisplayArea - displayAreaInsets: {"left":0,"top":59,"right":0,"bottom":34}
LOG  [2023-01-18T04:28:40.677Z] setDefaultDisplayArea - newDisplayArea: {"x":0,"y":0,"width":393,"height":852}
LOG  [2023-01-18T04:28:40.678Z] setDefaultDisplayArea - displayAreaOffset: {"x":0,"y":0}
LOG  [2023-01-18T04:28:40.679Z] [AdaptivePopover] getDisplayArea - displayAreaInsets: {"left":0,"top":59,"right":0,"bottom":34}
LOG  [2023-01-18T04:28:40.682Z] calculateRectFromRef - waiting for ref
LOG  [2023-01-18T04:28:40.682Z] calculateRectFromRef - waiting for ref to move
LOG  [2023-01-18T04:28:40.690Z] measureContent - new requestedContentSize: {"width":291.3333435058594,"height":151.333251953125} (used to be null)
LOG  [2023-01-18T04:28:40.693Z] handleChange - waiting 100ms to accumulate all changes
LOG  [2023-01-18T04:28:40.706Z] [AdaptivePopover] getDisplayArea - displayAreaInsets: {"left":0,"top":59,"right":0,"bottom":34}
LOG  [2023-01-18T04:28:40.706Z] measureContent - new requestedContentSize: {"width":291.3333435058594,"height":163} (used to be {"width":291.3333435058594,"height":151.333251953125})
LOG  [2023-01-18T04:28:40.710Z] handleChange - waiting 100ms to accumulate all changes
LOG  [2023-01-18T04:28:40.711Z] measureContent - new requestedContentSize: {"width":291.3333435058594,"height":174.6666259765625} (used to be {"width":291.3333435058594,"height":163})
LOG  [2023-01-18T04:28:40.714Z] handleChange - waiting 100ms to accumulate all changes
LOG  [2023-01-18T04:28:40.714Z] measureContent - Skipping, content size did not change
LOG  [2023-01-18T04:28:40.714Z] measureContent - new requestedContentSize: {"width":291.3333435058594,"height":169} (used to be {"width":291.3333435058594,"height":174.6666259765625})
LOG  [2023-01-18T04:28:40.717Z] handleChange - waiting 100ms to accumulate all changes
LOG  [2023-01-18T04:28:40.717Z] measureContent - new requestedContentSize: {"width":291.3333435058594,"height":174.6666259765625} (used to be {"width":291.3333435058594,"height":169})
LOG  [2023-01-18T04:28:40.721Z] handleChange - waiting 100ms to accumulate all changes
LOG  [2023-01-18T04:28:40.722Z] measureContent - new requestedContentSize: {"width":291.3333435058594,"height":172} (used to be {"width":291.3333435058594,"height":174.6666259765625})
LOG  [2023-01-18T04:28:40.725Z] handleChange - waiting 100ms to accumulate all changes
LOG  [2023-01-18T04:28:40.725Z] measureContent - new requestedContentSize: {"width":291.3333435058594,"height":174.6666259765625} (used to be {"width":291.3333435058594,"height":172})
LOG  [2023-01-18T04:28:40.728Z] handleChange - waiting 100ms to accumulate all changes
LOG  [2023-01-18T04:28:40.728Z] measureContent - new requestedContentSize: {"width":291.3333435058594,"height":173.333251953125} (used to be {"width":291.3333435058594,"height":174.6666259765625})
LOG  [2023-01-18T04:28:40.731Z] handleChange - waiting 100ms to accumulate all changes
LOG  [2023-01-18T04:28:40.731Z] measureContent - new requestedContentSize: {"width":291.3333435058594,"height":174.6666259765625} (used to be {"width":291.3333435058594,"height":173.333251953125})
LOG  [2023-01-18T04:28:40.735Z] handleChange - waiting 100ms to accumulate all changes
LOG  [2023-01-18T04:28:40.735Z] measureContent - new requestedContentSize: {"width":291.3333435058594,"height":151.333251953125} (used to be {"width":291.3333435058594,"height":174.6666259765625})
LOG  [2023-01-18T04:28:40.738Z] handleChange - waiting 100ms to accumulate all changes
LOG  [2023-01-18T04:28:40.840Z] handleChange - requestedContentSize: {"width":291.3333435058594,"height":151.333251953125}
LOG  [2023-01-18T04:28:40.840Z] handleChange - displayArea: {"x":0,"y":59,"width":393,"height":759}
LOG  [2023-01-18T04:28:40.840Z] handleChange - fromRect: {"x":10,"y":159.99999872843426,"width":152.3333282470703,"height":31.333332061767578}
LOG  [2023-01-18T04:28:40.840Z] handleChange - placement: "bottom"
LOG  [2023-01-18T04:28:40.842Z] computeGeometry - initial chosen geometry: {"popoverOrigin":{"x":10,"y":191.33333079020184},"anchorPoint":{"x":86.16666412353516,"y":191.33333079020184},"placement":"bottom","forcedContentSize":{"width":373,"height":616.6666692097981},"viewLargerThanDisplayArea":{"height":false,"width":false}}
LOG  [2023-01-18T04:28:40.842Z] computeGeometry - final chosen geometry: {"popoverOrigin":{"x":10,"y":191.33333079020184},"anchorPoint":{"x":86.16666412353516,"y":191.33333079020184},"placement":"bottom","forcedContentSize":{"width":373,"height":616.6666692097981},"viewLargerThanDisplayArea":{"height":false,"width":false}}
LOG  [2023-01-18T04:28:40.848Z] handleChange - animating in
LOG  [2023-01-18T04:28:40.848Z] getTranslateOrigin - popoverSize: {"width":291.3333435058594,"height":159.333251953125}
LOG  [2023-01-18T04:28:40.848Z] getTranslateOrigin - anchorPoint: {"x":86.16666412353516,"y":191.33333079020184}
LOG  [2023-01-18T04:28:40.850Z] animateIn - translateStart: {"x":-59.50000762939453,"y":1815.6667048136394}
LOG  [2023-01-18T04:28:40.850Z] animateIn - translatePoint: {"x":10,"y":191.33333079020184}
LOG  [2023-01-18T04:28:40.855Z] Setting up keyboard listeners
LOG  [2023-01-18T04:28:40.855Z] [AdaptivePopover] getDisplayArea - displayAreaInsets: {"left":0,"top":59,"right":0,"bottom":34}
LOG  [2023-01-18T04:28:40.857Z] [AdaptivePopover] getDisplayArea - displayAreaInsets: {"left":0,"top":59,"right":0,"bottom":34}
LOG  [2023-01-18T04:28:40.873Z] measureContent - Skipping, content size did not change
LOG  [2023-01-18T04:28:41.222Z] animateIn - onOpenComplete - Calculated Popover Rect: {"x":10,"y":199.3333740234375,"width":291.3333435058594,"height":151.333251953125}
LOG  [2023-01-18T04:28:41.222Z] animateIn - onOpenComplete - Calculated Arrow Rect: {"x":78.33333587646484,"y":191.3333740234375,"width":18,"height":10}
```
</details>
 
</details>

### CODE REVIEW

**Comment by hawkrives** [comment]
Do we need this? Or can we replace it with UIMenu / Paper.Menu?

**Comment by drewvolz** [comment]
- Current implementation will crash with current 4.x.
- Upgraded implementation fixes the crash and needs these changes
  
All issues in this PR are now resolved with v5.1.7 of the popover.
  
Upgrading to paper.menu or context menu would be cool, but a larger undertaking. Not sure how much of an effort or undertaking it would be for that to replace what we have.

File: modules/filter/filter-popover.tsx [hunk 1]
```diff
@@ -24,8 +26,12 @@ export function FilterPopover(props: Props): JSX.Element {
 			onRequestClose={() => onClosePopover(filter)}
 			placement={PopoverPlacement.BOTTOM}
 			popoverStyle={popoverContainer}
+			displayAreaInsets={safeareaInsets}
 		>
-			<FilterSection filter={filter} onChange={(filter) => setFilter(filter)} />
+			{/* This view wrapper shouldn't be needed but it does appear to fix a rendering issue */}
+			<View style={popoverContainer}>
```
Comment (hawkrives) [inline modules/filter/filter-popover.tsx:32]: Is this view still needed in 5.1.7?

**Review by hawkrives — APPROVED** [review]

File: modules/filter/filter-popover.tsx (hunk 1 above)
Comment (hawkrives) [reply modules/filter/filter-popover.tsx:32]: Also: "+  29:4  warning  Props should be sorted alphabetically  react/jsx-sort-props"

Comment (drewvolz) [reply modules/filter/filter-popover.tsx:32]: Still needed in 5.1.7. I'll sort the props.

**Pull request merged**